src/main/java/MealDemo/service/MealPlanningService.java

```diff
@@ -4,27 +4,18 @@
 import MealDemo.entity.Meal;
 
 import java.util.List;
-import java.util.Optional;
 
 public interface MealPlanningService {
 
-    // VIEW ENTITIES
     List<Meal> getMeals();
     List<Frequency> getFrequencies();
 
-    // FIND BY ID
-    Optional<Meal> findMealById(int mealId);
+    Meal findMealById(int mealId);
```

Comment:
Why did you get rid of the optional here? In doing so you might have broken `MealPlanningRESTController.findBymealById`